warehouse/clickhouse/clickhouse.go

```diff
@@ -0,0 +1,663 @@
+		if err != nil {
+			logger.Errorf("CH: Error in converting object location to object key for table:%s: %s,%v", tableName, objectLocation, err)
+			return nil, err
+		}
+		dirName := "/rudder-warehouse-load-uploads-tmp/"
+		tmpDirPath, err := misc.CreateTMPDIR()
+		if err != nil {
+			logger.Errorf("CH: Error in getting tmp directory for downloading load file for table:%s: %s, %v", tableName, objectLocation, err)
+			return nil, err
+		}
+		ObjectPath := tmpDirPath + dirName + fmt.Sprintf(`%s_%s_%d/`, ch.Warehouse.Destination.DestinationDefinition.Name, ch.Warehouse.Destination.ID, time.Now().Unix()) + object
+		err = os.MkdirAll(filepath.Dir(ObjectPath), os.ModePerm)
+		if err != nil {
+			logger.Errorf("CH: Error in making tmp directory for downloading load file for table:%s: %s, %s %v", tableName, objectLocation, err)
+			return nil, err
+		}
+		objectFile, err := os.Create(ObjectPath)
+		if err != nil {
+			logger.Errorf("CH: Error in creating file in tmp directory for downloading load file for table:%s: %s, %v", tableName, objectLocation, err)
+			return nil, err
+		}
+		downloader, err := filemanager.New(&filemanager.SettingsT{
+			Provider: warehouseutils.ObjectStorageType(ch.Warehouse.Destination.DestinationDefinition.Name, ch.Warehouse.Destination.Config),
+			Config:   ch.Warehouse.Destination.Config,
+		})
+		err = downloader.Download(objectFile, object)
+		if err != nil {
+			logger.Errorf("CH: Error in downloading file in tmp directory for downloading load file for table:%s: %s, %v", tableName, objectLocation, err)
+			return nil, err
+		}
+		fileName := objectFile.Name()
+		if err = objectFile.Close(); err != nil {
+			logger.Errorf("CH: Error in closing downloaded file in tmp directory for downloading load file for table:%s: %s, %v", tableName, objectLocation, err)
+			return nil, err
+		}
+		fileNames = append(fileNames, fileName)
+	}
+	return fileNames, nil
+
+}
+func generateArgumentString(arg string, length int) string {
+	var argString string
+	for i := 0; i < length; i++ {
+		if i == length-1 {
+			argString += arg
+		} else {
+			argString += arg + ","
+		}
+
+	}
+	return argString
+}
+
+// typeCastDataFromType typeCasts string data to the mentioned data type
+func typeCastDataFromType(data string, datatype string) interface{} {
+	switch datatype {
+	case "int":
+		i, err := strconv.Atoi(data)
+		if err != nil {
+			return nil
+		}
+		return i
+	case "float":
+		f, err := strconv.ParseFloat(data, 64)
+		if err != nil {
+			return nil
+		}
+		return f
+	case "datetime":
+		t, err := time.Parse(time.RFC3339, data)
+		if err != nil {
+			return ""
+		}
+		return t
+	case "boolean":
+		b, err := strconv.ParseBool(data)
+		if err != nil {
+			return nil
+		}
+		if b {
+			return 1
+		}
+		return 0
+	}
+	return data
+}
+
+// loadTable loads table to clickhouse from the load files
+func (ch *HandleT) loadTable(tableName string, columnMap map[string]string, forceLoad bool) (err error) {
+	if !forceLoad {
```

Comment:
we can remove forceLoad arg and code related to that as that was for other warehouses where we are loading users/identifies tables separately together